isisd/isis_tlvs.c

```diff
@@ -2681,9 +2681,14 @@ void isis_tlvs_add_ipv4_addresses(struct isis_tlvs *tlvs,
 {
 	struct listnode *node;
 	struct prefix_ipv4 *ip_addr;
+	unsigned int addr_count = 0;
 
-	for (ALL_LIST_ELEMENTS_RO(addresses, node, ip_addr))
+	for (ALL_LIST_ELEMENTS_RO(addresses, node, ip_addr)) {
 		isis_tlvs_add_ipv4_address(tlvs, &ip_addr->prefix);
+		addr_count++;
+		if (addr_count >= 63)
+			break;
```

Comment:
Addresses 64 and onwards do not get sent. It should be noted that this TLV is only relevant for usage as nexthop. For routing destinations, the IP reachability TLV will be included of which an unlimited count can occur.
So even including a single IP address will usually suffice given that we have unnumbered/onlink support.